qiskit_nature/second_q/transformers/basis_transformer.py

```diff
@@ -0,0 +1,97 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The basis transformer."""
+
+from __future__ import annotations
+from typing import cast
+
+from qiskit_nature.exceptions import QiskitNatureError
+from qiskit_nature.second_q.hamiltonians import ElectronicEnergy, Hamiltonian
+from qiskit_nature.second_q.operators import PolynomialTensor
+from qiskit_nature.second_q.operators.electronic_integrals import ElectronicIntegrals
+from qiskit_nature.second_q.problems import BaseProblem, ElectronicStructureProblem
+from qiskit_nature.second_q.properties.bases import ElectronicBasis
+
+from .base_transformer import BaseTransformer
+
+
+class BasisTransformer(BaseTransformer):
+    """TODO."""
+
+    # TODO: figure out how we can make its interface non-electronic specific
+    def __init__(
+        self,
+        initial_basis: ElectronicBasis,
+        final_basis: ElectronicBasis,
+        coefficients: ElectronicIntegrals,
```

Comment:
Given the above, I would like to raise suggestions/questions:
- Shall we have a general `Basis` Enum? Or shall we just unionize `ElectronicBasis` and `VibrationalBasis`? Or shall we leave this as is for now?
- `ElectronicIntegrals` is also custom to electronic stack. I think a better type hint here would be some `Protocol` which states that the object needs to support `.einsum`. Maybe `Einsummable`? That would fit for `ElectronicIntegrals` and `PolynomialTensor` as of now.

Reply:
`einsum` was not actually the real requirement here. Instead, we rely on `.transpose` to exist, so I have added a `Transposable` protocol which indicates its existence :+1: